src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts

```diff
@@ -191,6 +203,28 @@ function createMirrorNodeService() {
     }
   };
 
+  function decodeLog(signatureMap: Map<string, any>, logs: any[]) {
```

Comment:
I believe you can use the `MirrorNodeProposalEventLog` type for type safety here: 
`logs.forEach((log: MirrorNodeProposalEventLog) => ...`